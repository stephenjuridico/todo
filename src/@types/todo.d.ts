import {RouteProp} from '@react-navigation/native';
import {Priority, SortBy, Status} from '../constants/constants';
import type {NativeStackNavigationProp} from '@react-navigation/native-stack';

export interface ITodo {
  id: string;
  title: string;
  description: string;
  status: Status;
  priority: Priority;
  createdAt?: string;
  updatedAt?: string;
}

export interface IFilters {
  status: string;
  name: SortBy;
  priority: SortBy;
}

export interface Response {
  message: string;
  item?: ITodo;
  error?: any;
}

export type TodoContextType = {
  data: ITodo[];
  mapped: MappedTodoType;
  filters: IFilters;
  isLoading: boolean;
  createTodo: (todo: ITodo) => Promise<Response>;
  updateTodo: (todo: ITodo) => Promise<Response>;
  updateStatus: (
    id: string,
    status: Status,
    showToast?: boolean,
  ) => Promise<Response>;
  setFilters: (filters: IFilters) => void;
};

export type MappedTodoType = {[key: string]: {count: number; items: ITodo[]}};

export type RootStackParamList = {
  Main: undefined;
  CreateTask: {item: ITodo} | undefined;
};

export type RootRouteProps<RouteName extends keyof RootStackParamList> =
  RouteProp<RootStackParamList, RouteName>;

export type MainScreenNavigationProp = NativeStackNavigationProp<
  RootStackParamList,
  'Main'
>;
