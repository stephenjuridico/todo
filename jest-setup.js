jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock'),
);

// Fake timers using Jest
beforeEach(() => {
  jest.useFakeTimers();
});
