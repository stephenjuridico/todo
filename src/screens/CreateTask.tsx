import React, {FC} from 'react';
import {Formik} from 'formik';
import {
  Button,
  FormControl,
  Heading,
  Input,
  Stack,
  Icon,
  HStack,
  Select,
  CheckIcon,
  WarningOutlineIcon,
  TextArea,
  ScrollView,
  VStack,
} from 'native-base';
import Feather from 'react-native-vector-icons/Feather';
import {
  Priority,
  PriorityList,
  Status,
  StatusList,
} from '../constants/constants';
import * as Yup from 'yup';
import NavBar from '../components/NavBar';
import {useNavigation, useRoute} from '@react-navigation/native';
import {useTodo} from '../context/TodoContext';
import * as uuid from 'uuid';
import {successToast} from '../utils/useToast';
import {RootRouteProps} from '../@types/todo';

const CreateTask: FC = () => {
  const {isLoading, createTodo, updateTodo} = useTodo();
  const navigation = useNavigation();
  const routes = useRoute<RootRouteProps<'CreateTask'>>();
  const TodoSchema = Yup.object().shape({
    title: Yup.string().required('Required Field'),
    description: Yup.string().required('Required Field'),
    status: Yup.string().required('Required Field'),
    priority: Yup.number().required('Required Field'),
  });

  const todoItem = routes.params?.item;
  const isEditing = todoItem ? true : false;

  return (
    <VStack bg="white" flex={1} safeArea>
      <NavBar />
      <Heading px={4} mt={4}>
        {isEditing ? 'Update Task' : 'New Task'}
      </Heading>
      <Formik
        initialValues={{
          title: todoItem?.title || '',
          description: todoItem?.description || '',
          status: todoItem?.status || Status.Pending,
          priority: todoItem?.priority || Priority.None,
        }}
        validationSchema={TodoSchema}
        onSubmit={async values => {
          try {
            let response = null;
            if (isEditing && todoItem) {
              response = await updateTodo({
                ...todoItem,
                ...values,
              });
            } else {
              response = await createTodo({
                id: uuid.v4(),
                ...values,
              });
            }
            console.log('Item', response);
            successToast('Success', response.message);
            navigation.goBack();
          } catch (err) {
            console.log('err', err);
          }
        }}>
        {({
          errors,
          values,
          touched,
          handleChange,
          handleBlur,
          isValid,
          setFieldValue,
          handleSubmit,
        }) => (
          <ScrollView px={4}>
            <Stack mt={4} space={4}>
              <FormControl
                isInvalid={errors.title && touched.title ? true : false}>
                <FormControl.Label _text={{fontSize: 'xs'}}>
                  Title
                </FormControl.Label>
                <Input
                  testID="inputTitle"
                  placeholder=""
                  onChangeText={handleChange('title')}
                  onBlur={handleBlur('title')}
                  value={values.title}
                />
                <FormControl.ErrorMessage>
                  {errors.title}
                </FormControl.ErrorMessage>
              </FormControl>
              <FormControl
                isInvalid={
                  errors.description && touched.description ? true : false
                }>
                <FormControl.Label _text={{fontSize: 'xs'}}>
                  Description
                </FormControl.Label>
                {/* missing autoCompleteType - https://github.com/GeekyAnts/NativeBase/issues/5438 */}
                <TextArea
                  testID="inputDescription"
                  numberOfLines={4}
                  placeholder=""
                  onChangeText={handleChange('description')}
                  onBlur={handleBlur('description')}
                  value={values.description}
                  autoCompleteType={undefined}
                />
                <FormControl.ErrorMessage>
                  {errors.description}
                </FormControl.ErrorMessage>
              </FormControl>
              {isEditing && (
                <FormControl>
                  <FormControl.Label>Status</FormControl.Label>
                  <Select
                    testID="inputStatus"
                    accessibilityLabel="Choose Status"
                    placeholder="Choose Status"
                    selectedValue={values.status}
                    onValueChange={val => setFieldValue('status', val)}
                    _selectedItem={{
                      // bg: 'primary.600',
                      endIcon: <CheckIcon size={5} />,
                    }}
                    mt="1">
                    {StatusList.map(status => {
                      return (
                        <Select.Item
                          key={status.value}
                          label={status.label}
                          value={status.value}
                        />
                      );
                    })}
                  </Select>
                  <FormControl.ErrorMessage
                    leftIcon={<WarningOutlineIcon size="xs" />}>
                    Please make a selection!
                  </FormControl.ErrorMessage>
                </FormControl>
              )}
              <FormControl>
                <FormControl.Label _text={{fontSize: 'xs'}}>
                  Priority
                </FormControl.Label>
                <HStack space={2} mt={2} justifyContent={'space-between'}>
                  {PriorityList.map(v => {
                    const selected = v.value === values.priority;
                    return (
                      <Button
                        testID={`btnPriority${v.label}`}
                        key={v.value}
                        onPress={() => setFieldValue('priority', v.value)}
                        variant={selected ? 'outline' : 'ghost'}
                        size={'sm'}
                        colorScheme={v.color}
                        // color={v.color}
                        borderColor={`${v.color}.500`}
                        leftIcon={<Icon as={Feather} name="flag" />}>
                        {v.label}
                      </Button>
                    );
                  })}
                </HStack>
              </FormControl>
              <Button
                testID="btnSubmit"
                mt={'8'}
                isLoading={isLoading}
                isDisabled={!isValid}
                onPress={handleSubmit}>
                {isEditing ? 'UPDATE' : 'CREATE'}
              </Button>
            </Stack>
          </ScrollView>
        )}
      </Formik>
    </VStack>
  );
};

export default CreateTask;
