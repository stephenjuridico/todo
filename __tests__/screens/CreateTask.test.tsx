import 'react-native';
import React from 'react';
import {fireEvent, render, screen, waitFor} from '../../src/test/test-utils';
import CreateTask from '../../src/screens/CreateTask';
import {ITodo, TodoContextType} from '../../src/@types/todo';
import {
  TodoProvider,
  initialFiltersValue,
  initialMappedValue,
} from '../../src/context/TodoContext';
import {mockedRouteParam} from '../../jest-setup';

describe('CreateTask', () => {
  const expectedItem = {
    id: '123123',
    title: 'Test',
    description: 'Test Description',
    priority: 3,
    status: 'pending',
  } as ITodo;

  const mockedCreateTodo = jest.fn();
  const mockedUpdateTodo = jest.fn();

  const contextValues = {
    data: [],
    mapped: initialMappedValue,
    isLoading: false,
    createTodo: mockedCreateTodo,
    updateTodo: mockedUpdateTodo,
    updateStatus: jest.fn(),
    filters: initialFiltersValue,
    setFilters: jest.fn(),
  } as TodoContextType;

  const renderCreateTask = () => {
    render(
      <TodoProvider value={contextValues}>
        <CreateTask />
      </TodoProvider>,
    );
  };

  it('renders correctly', () => {
    renderCreateTask();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('show inputs correctly', () => {
    renderCreateTask();
    const title = screen.getByText('New Task');
    const inputTitle = screen.getByTestId('inputTitle');
    const inputDescription = screen.getByTestId('inputDescription');
    const inputStatus = screen.queryByTestId('inputStatus');
    const btnPriorityHigh = screen.getByTestId('btnPriorityHigh');
    const btnPriorityMedium = screen.getByTestId('btnPriorityMedium');
    const btnPriorityLow = screen.getByTestId('btnPriorityLow');
    const btnPriorityNone = screen.getByTestId('btnPriorityNone');
    const btnSubmit = screen.getByTestId('btnSubmit');

    expect(title).toBeTruthy();
    expect(inputTitle).toBeTruthy();
    expect(inputDescription).toBeTruthy();
    expect(inputStatus).toBeFalsy();
    expect(btnPriorityHigh).toBeTruthy();
    expect(btnPriorityMedium).toBeTruthy();
    expect(btnPriorityLow).toBeTruthy();
    expect(btnPriorityNone).toBeTruthy();
    expect(btnSubmit).toBeTruthy();
  });

  it('shows required fields', async () => {
    renderCreateTask();
    const btnSubmit = screen.getByTestId('btnSubmit');
    fireEvent.press(btnSubmit);

    const required = await screen.findAllByText('Required Field');
    // Only to required fields (Title and Description)
    expect(required).toHaveLength(2);
  });

  // it('create task', async () => {
  //   renderCreateTask();
  //   const inputTitle = screen.getByTestId('inputTitle');
  //   const inputDescription = screen.getByTestId('inputDescription');
  //   const btnSubmit = screen.getByTestId('btnSubmit');

  //   fireEvent.changeText(inputTitle, expectedItem.title);
  //   fireEvent.changeText(inputDescription, expectedItem.description);

  //   expect(inputTitle).toMatch(expectedItem.title);
  //   expect(inputDescription).toMatch(expectedItem.description);

  //   fireEvent.press(btnSubmit);
  //   // Only to required fields (Title and Description)
  //   expect(mockedCreateTodo).toBeCalled();
  // });

  // it('edit task and show prefilled inputs correctly', () => {
  //   mockedRouteParam.mockReturnValue({params: {item: expectedItem}});
  //   renderCreateTask();
  //   const title = screen.getByText('New Task');
  //   const inputTitle = screen.getByTestId('inputTitle');
  //   const inputDescription = screen.getByTestId('inputDescription');
  //   const inputStatus = screen.queryByTestId('inputStatus');
  //   const btnPriorityHigh = screen.getByTestId('btnPriorityHigh');
  //   const btnPriorityMedium = screen.getByTestId('btnPriorityMedium');
  //   const btnPriorityLow = screen.getByTestId('btnPriorityLow');
  //   const btnPriorityNone = screen.getByTestId('btnPriorityNone');
  //   const btnSubmit = screen.getByTestId('btnSubmit');

  //   expect(title).toBeTruthy();
  //   expect(inputTitle).toBeTruthy();
  //   expect(inputDescription).toBeTruthy();
  //   expect(inputStatus).toBeTruthy();
  //   expect(btnPriorityHigh).toBeTruthy();
  //   expect(btnPriorityMedium).toBeTruthy();
  //   expect(btnPriorityLow).toBeTruthy();
  //   expect(btnPriorityNone).toBeTruthy();
  //   expect(btnSubmit).toBeTruthy();
  // });
});
