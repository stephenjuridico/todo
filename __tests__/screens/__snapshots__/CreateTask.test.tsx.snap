// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTask renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={{}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderRadius": 9999,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        testID="btnGoBack"
      >
        <Text
          allowFontScaling={false}
          dataSet={{}}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "color": "#e6b200",
                "fontSize": 24,
                "height": 24,
                "lineHeight": 24,
                "width": 24,
              },
              {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          ÔÑê
        </Text>
      </View>
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <Text
      dataSet={{}}
      style={
        {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 24,
          "fontStyle": "normal",
          "fontWeight": "700",
          "letterSpacing": 0,
          "lineHeight": 30,
          "marginTop": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "textDecorationLine": undefined,
        }
      }
    >
      New Task
    </Text>
    <View
      dataSet={{}}
      style={{}}
    />
    <RCTScrollView
      contentContainerStyle={
        [
          {},
          {
            "dataSet": {},
          },
          {},
        ]
      }
      dataSet={{}}
      style={
        {
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <View>
        <View
          dataSet={{}}
          style={
            {
              "marginTop": 16,
            }
          }
        >
          <View
            dataSet={{}}
            isInvalid={false}
            style={
              {
                "width": "100%",
              }
            }
          >
            <View
              dataSet={{}}
              feedbackId="field-4-feedback"
              hasFeedbackText={true}
              hasHelpText={false}
              helpTextId="field-4-helptext"
              isDisabled={false}
              isInvalid={false}
              isReadOnly={false}
              isRequired={false}
              labelId="field-4-label"
              nativeID="field-4-label"
              setHasFeedbackText={[Function]}
              setHasHelpText={[Function]}
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginBottom": 4,
                  "marginTop": 4,
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#737373",
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "500",
                    "letterSpacing": 0,
                    "lineHeight": 18,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Title
              </Text>
            </View>
            <View
              dataSet={{}}
              isFocused={false}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#d4d4d4",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "overflow": "hidden",
                }
              }
            >
              <TextInput
                accessibilityDescribedBy="field-4-feedback"
                accessible={true}
                dataSet={{}}
                disabled={false}
                editable={true}
                isRequired={false}
                nativeID="field-4-input"
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                placeholder=""
                placeholderTextColor="#a3a3a3"
                readOnly={false}
                required={false}
                secureTextEntry={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  }
                }
                testID="inputTitle"
                value=""
              />
            </View>
          </View>
          <View
            dataSet={{}}
            style={
              {
                "height": 16,
              }
            }
          />
          <View
            dataSet={{}}
            isInvalid={false}
            style={
              {
                "width": "100%",
              }
            }
          >
            <View
              dataSet={{}}
              feedbackId="field-5-feedback"
              hasFeedbackText={true}
              hasHelpText={false}
              helpTextId="field-5-helptext"
              isDisabled={false}
              isInvalid={false}
              isReadOnly={false}
              isRequired={false}
              labelId="field-5-label"
              nativeID="field-5-label"
              setHasFeedbackText={[Function]}
              setHasHelpText={[Function]}
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginBottom": 4,
                  "marginTop": 4,
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#737373",
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "500",
                    "letterSpacing": 0,
                    "lineHeight": 18,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Description
              </Text>
            </View>
            <View
              dataSet={{}}
              isFocused={false}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "#d4d4d4",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "height": 80,
                  "overflow": "hidden",
                }
              }
            >
              <TextInput
                accessibilityDescribedBy="field-5-feedback"
                accessible={true}
                dataSet={{}}
                disabled={false}
                editable={true}
                isRequired={false}
                multiline={true}
                nativeID="field-5-input"
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onKeyPress={[Function]}
                placeholder=""
                placeholderTextColor="#a3a3a3"
                readOnly={false}
                required={false}
                secureTextEntry={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  }
                }
                testID="inputDescription"
                textAlignVertical="top"
                value=""
              />
            </View>
          </View>
          <View
            dataSet={{}}
            style={
              {
                "height": 16,
              }
            }
          />
          <View
            dataSet={{}}
            style={
              {
                "width": "100%",
              }
            }
          >
            <View
              dataSet={{}}
              feedbackId="field-3-feedback"
              hasFeedbackText={false}
              hasHelpText={false}
              helpTextId="field-3-helptext"
              isDisabled={false}
              isInvalid={false}
              isReadOnly={false}
              isRequired={false}
              labelId="field-3-label"
              nativeID="field-3-label"
              setHasFeedbackText={[Function]}
              setHasHelpText={[Function]}
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginBottom": 4,
                  "marginTop": 4,
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#737373",
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "500",
                    "letterSpacing": 0,
                    "lineHeight": 18,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Priority
              </Text>
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 8,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="red"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#ef4444",
                    "borderRadius": 9999,
                    "color": "#ffc600",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                  }
                }
                testID="btnPriorityHigh"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <Text
                    allowFontScaling={false}
                    dataSet={{}}
                    selectable={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        {
                          "color": "#dc2626",
                          "fontSize": 16,
                          "height": 16,
                          "lineHeight": 16,
                          "width": 16,
                        },
                        {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ÔÖØ
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "width": 6,
                      }
                    }
                  />
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#dc2626",
                          "fontFamily": undefined,
                          "fontSize": 12,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 18,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      High
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 8,
                  }
                }
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="amber"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#f59e0b",
                    "borderRadius": 9999,
                    "color": "#ffc600",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                  }
                }
                testID="btnPriorityMedium"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <Text
                    allowFontScaling={false}
                    dataSet={{}}
                    selectable={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        {
                          "color": "#d97706",
                          "fontSize": 16,
                          "height": 16,
                          "lineHeight": 16,
                          "width": 16,
                        },
                        {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ÔÖØ
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "width": 6,
                      }
                    }
                  />
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#d97706",
                          "fontFamily": undefined,
                          "fontSize": 12,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 18,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Medium
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 8,
                  }
                }
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="green"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#22c55e",
                    "borderRadius": 9999,
                    "color": "#ffc600",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                  }
                }
                testID="btnPriorityLow"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <Text
                    allowFontScaling={false}
                    dataSet={{}}
                    selectable={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        {
                          "color": "#16a34a",
                          "fontSize": 16,
                          "height": 16,
                          "lineHeight": 16,
                          "width": 16,
                        },
                        {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ÔÖØ
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "width": 6,
                      }
                    }
                  />
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#16a34a",
                          "fontFamily": undefined,
                          "fontSize": 12,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 18,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Low
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 8,
                  }
                }
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="gray"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#71717a",
                    "borderRadius": 9999,
                    "borderWidth": 2,
                    "color": "#ffc600",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                  }
                }
                testID="btnPriorityNone"
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <Text
                    allowFontScaling={false}
                    dataSet={{}}
                    selectable={false}
                    style={
                      [
                        {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        {
                          "color": "#52525b",
                          "fontSize": 16,
                          "height": 16,
                          "lineHeight": 16,
                          "width": 16,
                        },
                        {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    ÔÖØ
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "width": 6,
                      }
                    }
                  />
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#52525b",
                          "fontFamily": undefined,
                          "fontSize": 12,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 18,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      None
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            dataSet={{}}
            style={
              {
                "height": 16,
              }
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            colorScheme="primary"
            dataSet={{}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ffc600",
                "borderRadius": 9999,
                "color": "#fff",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 32,
                "paddingBottom": 10,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 10,
              }
            }
            testID="btnSubmit"
          >
            <View
              dataSet={{}}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
              test={true}
            >
              <View
                dataSet={{}}
                style={{}}
              >
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#fafafa",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  CREATE
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
